### PR TITLE
Include VK_LAYER_KHRONOS_timeline_semaphore with CMake Linux.

### PR DESCRIPTION
**Pull request by ScottTodd**
Using the Vulkan Loader for now, see [these docs](https://vulkan.lunarg.com/doc/view/1.1.70.1/windows/loader_and_layer_interface.html#user-content-layer-discovery). We're also discussing ways to inject the layer ourselves, outside of the loader. That will reduce binary size and deployment complexity.

Bazel is up next.

Windows isn't yet supported by the layer.

### CODE REVIEW

File: iree/hal/vulkan/dynamic_symbols.cc [hunk 1]
```diff
@@ -132,22 +133,43 @@ StatusOr<ref_ptr<DynamicSymbols>> DynamicSymbols::Create(
...
+// TODO(scotttodd): Layer discovery on Windows
+#else
+  const char* vk_layer_path = ::getenv("VK_LAYER_PATH");
+  std::string combined_path;
+  if (vk_layer_path) {
+    combined_path = absl::StrCat(vk_layer_path, ":", vk_layer_path_extras);
+  } else {
+    combined_path = vk_layer_path_extras;
+  }
+  ::setenv("VK_LAYER_PATH", combined_path.c_str(), 0);
```
Comment (benvanik) [inline iree/hal/vulkan/dynamic_symbols.cc]: *shudder*

File: iree/hal/vulkan/CMakeLists.txt [hunk 2]
```diff
@@ -232,6 +233,25 @@ iree_cc_library(
   PUBLIC
 )
 
+if(${CMAKE_HOST_SYSTEM_NAME} STREQUAL "Windows")
```
Comment (benvanik) [inline iree/hal/vulkan/CMakeLists.txt]: can this be moved to build_tools/third_party/blah? It'd be nice to keep the cmake files we have mostly just declarative usage

File: iree/hal/vulkan/dynamic_symbols.cc (hunk 1 above)
Comment (ScottTodd) [reply iree/hal/vulkan/dynamic_symbols.cc]: Yeah >___> ...

The `VK_ICD_FILENAMES` code is used for SwiftShader (mostly Google upstream where it's harder to set environment variables outside of a program). This new code for `VK_LAYER_PATH`  is used across environments for `VK_LAYER_KHRONOS_timeline_semaphore`. It's gross and I'd rather inject the layer into our binaries somehow, without going through the loader.

File: iree/hal/vulkan/CMakeLists.txt (hunk 2 above)
Comment (ScottTodd) [reply iree/hal/vulkan/CMakeLists.txt]: Yep! I had the same thought but wasn't quite sure where to put it. How's this?

**Pull request closed without merging**